### DISCUSSION TITLE
incomplete-app-router-transaction

### DISCUSSION THREAD

**Discussion by chukusuccess** [vercel/next.js#70622] [App Router]
what does this error mean? it's caught by sentry when a sign-in attempt is made by some users and i can't make much sense of it. Also i have tried to upload source maps so i can see the bad code but I get this "Missing Source Context" error on sentry. Is this a NextJs 14.2.7 issue? I need some help, suggestions, and answers please.🙏

**Comment by aldenquimby** [comment]
@chukusuccess are you using Clerk for auth? We are seeing the same problem after Clerk sign in, with the same unhelpful Sentry log. The app hard crashes for users when this happens. Wondering if it may be a Clerk bug

**Reply by martingalovic** [reply]
Yes, same here, using Clerk as well

**Reply by DawnMD** [reply]
@aldenquimby getting the same error when using Auth0 as well. Weirdly I cannot reproduce this in local. Wondering how you can see the app hard crash?

**Reply by aldenquimby** [reply]
@DawnMD we use Sentry for error monitoring, and it has a replays feature where you can watch user sessions. Every time this error happens, users land on our "app hard crashed" page

**Reply by DawnMD** [reply]
@aldenquimby Hmm, interesting. For me it's not crashing the page. Or might by our sentry not catching the latest replays as quota is not available. Let see if we can replicate the same in local!

**Reply by jescalan** [reply]
👀 clerk employee, we're looking into this one

**Reply by elie222** [reply]
I'm facing the same issue, but not using Clerk. Hard crash as well.

**Reply by elie222** [reply]
@jescalan 
Did you figure out the issue for Clerk? Facing this issue too, but can't find the root of it

**Comment by chukusuccess** [comment]
@aldenquimby @DawnMD @martingalovic Sorry for the late response, been caught up with work but I was able to debug my issue. In my case the data that was supposed to be passed into the dashboard component after sign-in was sometimes incomplete (especially image links) so the navigation from sign-in page into the dashboard was breaking. After doing proper checks for data before rendering anything (especially Image tags), I stopped seeing the error on sentry. @aldenquimby I'm not using Clerk for auth in my project but you should check for the presence of data in the page you are routing to and see if that helps with the "incomplete" routing bug.

**Reply by DawnMD** [reply]
I was suspecting the same thing. But how do you differentiate when the images or maybe some fetch done loading?

**Reply by aldenquimby** [reply]
@chukusuccess could you give a rough before/after example of your next Image usage showing what the problem was and how you fixed it?

I appreciate the tip. I am seeing this when redirecting from Clerk sign-in to a dozen other pages, so I suspect I have a different problem, but am investigating.

**Comment by elie222** [comment]
Any updates here? This has been a major issue for us and just can't find a fix for it. Our issue isn't Clerk related.

**Comment by lsharples7** [comment]
Also running into this issue- using Firebase for auth. Hard to recreate (have been unable to repro locally) but happens at least once a day in prod. Usually comes with sentry message "Unexpected end of input" or "Invalid or unexpected token", though breadcrumbs show 200 responses. We have been telling users to log in with an incognito browser if they get the hard crash which seems to solve it, but not sure if it is actually 'solving' it or is just a transient issue.

Using next version 13.4.10 if relevant. Tried switching from router.push() to router.replace() when navigating from login to home page, but still seeing the issue.

Going to take a look at the proper checks for data rendering/image tags solution mentioned above, but would appreciate any other insight.

**Comment by djshubs** [comment]
we aren't using clerk, but we are seeing this being reported a lot on Sentry as well. 

I haven't validated why its happening for us, but my gut reaction is that when we're redirecting or pushing people to notFound() function that it causes this issue. 

My understanding is that the [redirect](https://nextjs.org/docs/app/api-reference/functions/redirect#server-component) and [notFound()](https://nextjs.org/docs/app/api-reference/functions/not-found#notfound) functions in nextjs are "throwing an error" which terminates the rendering. 

Any help or workaround would be appreciated.

**Reply by elie222** [reply]
There are other GitHub issues open around redirect throwing an error. Sentry handles it fine. It's not related to this issue.

**Comment by ozandalgali** [comment]
I am using Supabase yet also experiencing this issue on production, especially on some Android 10 devices, the points we use router hook we get this error often at least twice a day. I think I will leave router once and for all build separate pages for every route.

**Comment by ryananhtuan-nguyen** [comment]
Has anyone figured out what the issue is yet? I'm using Clerk and Supabase all together but those dont seem to be related to the issues.
This error only show up in Sentry, have not been able to reprod in local.

**Reply by elie222** [reply]
Nope. Would love some feedback on this issue too :(

**Reply by justman00** [reply]
same thing here, completely in the dark with this one

**Reply by ryananhtuan-nguyen** [reply]
How did you guys sending error to sentry? With a global-error.tsx file?
The incomplete-app-router-transaction for me comming from errors that not handled by the error.tsx file (so it is not just one but multiple error that referencing the same name "incomplete-app-router-transaction"). These errors for my case dont send users to the hard crash page (app is still functioning) but throwing error behind the scene. So I figure it has nothing to do with the error.

**Reply by albertocubeddu** [reply]
Any update? Just got the error for the first time; 
The change we've done is adding a UseContext in the page.

**Comment by ariannargesi** [comment]
Same issue here

**Comment by curiousYi** [comment]
`incomplete-app-router-transaction` is a fallback transaction name used by the Sentry SDK when it detects a router interaction but cannot yet determine the exact navigation context or transaction name. This can occur if there are rapid route changes or when the application hasn't fully resolved the route context at the time of the SDK's instrumentation. 

See [here](https://github.com/getsentry/sentry-javascript/blob/8dd8ac556bb429f57ab77475575fbc6520191cdb/packages/nextjs/src/client/routing/appRouterRoutingInstrumentation.ts#L10)

**Reply by jetaggart** [reply]
Is it safe to ignore this error?

**Comment by MattisAbrahamsson** [comment]
We started seeing this issue in Sentry on Feb 13th, since then we are getting around 1.5k errors in Sentry each day from this. I will ignore it for now since it doesn't appear to be affecting the user at all, but would be nice to not have to ignore so many Next.js related errors.

**Comment by ms-rushi** [comment]
I am also facing the same issue. Hundreds of unclear errors are being recorded in the sentry with label 
incomplete-app-router-transaction. What can be the possible reasons for the same and Is there any way to fix this or avoid getting these errors recorded.

**Comment by mertohancan** [comment]
Should we update our version to the v15?

**Reply by mannasdev** [reply]
that doesn't seem to solve the problem :(

**Reply by MattisAbrahamsson** [reply]
v15.2 has solved this error completely for us

**Reply by benjick** [reply]
Seeing this in 15.2.3

**Comment by ChandraShekar-22** [comment]
Any update on this issue. How to fix this?

**Comment by qoa11a** [comment]
I've been encountering this error since upgrading to `next@15.3.2` and `@sentry/nextjs@8.34.0`. At this point, I’m unsure of the root cause or how to resolve it beyond what's already been discussed in this thread.

I'd appreciate any insights or clarification.
